code/game/atoms_movable.dm

```diff
@@ -322,7 +333,7 @@
 /atom/movable/proc/update_bounds(olddir, newdir)
 	SEND_SIGNAL(src, COMSIG_MOVABLE_UPDATE_BOUNDS, args)
 
-	if(bound_width == bound_height && !bound_x && !bound_y) // We're a square and have no offset
+	if(bound_width == bound_height && !bound_x && !bound_y || newdir == olddir) // We're a square and have no offset or the direction hasn't changed

```

Comment:
It's probably best you make a second early return for the dir check to avoid confusion